CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+
+option(BUILD_TESTS "Build tests" OFF)
+option(BUILD_BENCHMARKS "Build benchmarks" OFF)
+option(USE_VCPKG "Use vcpkg for dependencies" OFF)
+
+if(COMMAND cmake_policy)
+    cmake_policy(SET CMP0003 NEW)
+endif(COMMAND cmake_policy)
+
+if (USE_VCPKG)
+    set(CMAKE_TOOLCHAIN_FILE ${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake
+    CACHE STRING "Vcpkg toolchain file")
+
+    if (BUILD_TESTS)
+        list(APPEND VCPKG_MANIFEST_FEATURES "tests")
+    endif()
+
+    if (BUILD_BENCHMARKS)
+        list(APPEND VCPKG_MANIFEST_FEATURES "benchmarks")
+    endif()
+endif()
+
+
+project(cachemere)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+find_package(Boost REQUIRED)
+
+add_library(cachemere INTERFACE)
+
+if ((CMAKE_CXX_COMPILER_ID STREQUAL "Clang") OR (CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
+  # using Clang or GCC
+  add_compile_options("-Wall" "-Wextra")
+elseif (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+  # using MSVC
+  add_compile_options("/W4")
+endif()
+
+target_link_libraries(cachemere
+    INTERFACE
+        Boost::boost
+)
+
+target_include_directories(cachemere
+    INTERFACE
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+        $<INSTALL_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
+)
```

Comment:
[important] The install path should be relative (it will be relative to the `CMAKE_INSTALL_PREFIX`). Also, I notice that you do not provide a so-called [Config file](https://cmake.org/cmake/help/v3.16/module/CMakePackageConfigHelpers.html). Doing so will help people reuse your project simply by using `target_link_libraries(myproject PUBLIC cachemere::cachemere)`. More and more libraries provide these files now (for instance, the AWS SDK does).

Below is what I suggest. You can test it with

```shell
mkdir debug
cd debug
cmake ..
cmake --install . --prefix $PWD/installed
```

which is what I did. The magic will then happen in the file `$PWD/installed/share/cachemere/cachemereConfig.cmake`, which will include `$PWD/installed/share/cachemere/cachemere-targets.cmake`.

So, my suggestion:

```suggestion
include(GNUInstallDirs)
target_include_directories(cachemere
    INTERFACE
        $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
        $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
)

set(INSTALL_CMAKEDIR "${CMAKE_INSTALL_DATAROOTDIR}/cachemere")
install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
install(TARGETS cachemere EXPORT cachemere-targets)
install(EXPORT cachemere-targets
    DESTINATION "${INSTALL_CMAKEDIR}"
    NAMESPACE cachemere::
)

include(CMakePackageConfigHelpers)
configure_package_config_file(cachemereConfig.cmake.in
    ${PROJECT_BINARY_DIR}/cachemereConfig.cmake
    INSTALL_DESTINATION "${INSTALL_CMAKEDIR}"
)
install(FILES ${PROJECT_BINARY_DIR}/cachemereConfig.cmake DESTINATION ${INSTALL_CMAKEDIR})
```

And you'll need to add the file `cachemereConfig.cmake.in` at the root of the repository which should contain

```cmake
@PACKAGE_INIT@

if(NOT TARGET cachemere::cachemere)
    include(CMakeFindDependencyMacro)
    find_dependency(Boost)
    include(${CMAKE_CURRENT_LIST_DIR}/cachemere-targets.cmake)
endif()
```